<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Estef Vazquez" />

<meta name="date" content="2025-04-04" />

<title>Figure 2 - GSEA</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ulceration_AM</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Figure 2 - GSEA</h1>
<h4 class="author">Estef Vazquez</h4>
<h4 class="date">2025-04-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Ulceration_paper_github/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250330code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250330)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250330code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250330)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong248524c">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 248524c
</a>
</p>
</div>
<div id="strongRepositoryversionstrong248524c"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 248524c.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/DE_results.rds
    Untracked:  data/DE_results_ranked.rds
    Untracked:  data/DE_results_ulceration_rankednew.rds
    Untracked:  data/annotation.rds
    Untracked:  data/clinical_am_prim.csv
    Untracked:  data/collectri_network_omnipath.rds
    Untracked:  data/normalized_counts.rds
    Untracked:  data/rawcounts_am.rds
    Untracked:  omnipathr-log/
    Untracked:  output/ulceration_combined_panel.pdf
    Untracked:  output/ulceration_mitotic.pdf
    Untracked:  output/volcanoplot.pdf
    Untracked:  volcanoplot.pdf

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/_site.yml
    Deleted:    analysis/teeth.Rmd
    Deleted:    data/teeth.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/figure2_GSEA.Rmd</code>) and HTML
(<code>docs/figure2_GSEA.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
248524c
</td>
<td>
Estef Vazquez
</td>
<td>
2025-04-04
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
dd1d8cb
</td>
<td>
Estef Vazquez
</td>
<td>
2025-04-04
</td>
<td>
wflow_rename("analysis/test_render_GSEA_TF.Rmd",
"analysis/figure2_GSEA.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
dd1d8cb
</td>
<td>
Estef Vazquez
</td>
<td>
2025-04-04
</td>
<td>
wflow_rename("analysis/test_render_GSEA_TF.Rmd",
"analysis/figure2_GSEA.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>We perform Gene Set Enrichment Analysis (GSEA) using GO terms and
PROGENy to identify pathway activity in ulcerated vs non-ulcerated acral
melanoma samples.</p>
<pre class="r"><code># Load required libraries
library(tidyverse)  
library(clusterProfiler)
library(enrichplot)
library(DOSE)
library(biomaRt)
library(org.Hs.eg.db)
library(progeny)
library(decoupleR)
library(here)</code></pre>
<div id="c---2d---gsea" class="section level2">
<h2>2C - 2D - GSEA</h2>
<pre class="r"><code># Data loading
# Load differential expression results pre-ranked for GSEA
ranked_GSEA &lt;- readRDS(&quot;data/DE_results_ranked.rds&quot;)
ranked_GSEA &lt;- rownames_to_column(ranked_GSEA, var = &quot;ENSEMBL_GENE_ID&quot;)

# Load gene annotation
gene_ann &lt;- readRDS(&quot;data/annotation.rds&quot;)
#gene_ann &lt;- rownames_to_column(gene_ann, var = &quot;ENSEMBL_GENE_ID&quot;)

# Add gene symbols
ranked_idmatch &lt;- inner_join(ranked_GSEA, gene_ann, by=&quot;ENSEMBL_GENE_ID&quot;)

# Make rownames unique
names &lt;- make.unique(ranked_idmatch$external_gene_name)
rownames(ranked_idmatch) &lt;- names

# Extract LFC and create a named vector for GSEA
geneList_FC &lt;- ranked_idmatch[,3]
names(geneList_FC) &lt;- as.character(ranked_idmatch[,1]) # Use ENSEMBL IDs as names

# Sort gene list in decreasing order by LFC
geneList_FC_ordered &lt;- sort(geneList_FC, decreasing = TRUE)

# ---GSEA ---
# Perform Gene Set Enrichment Analysis using GO Biological Process 
gsea_BP &lt;- gseGO(geneList = geneList_FC_ordered, 
             ont =&quot;BP&quot;, 
             keyType = &quot;ENSEMBL&quot;, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db,
             pAdjustMethod = &quot;BH&quot;,
             eps = 0) 

go_BP_df &lt;- (as.data.frame(gsea_BP))

# saveRDS(gsea_BP, &quot;results/GSEA_BP_results.rds&quot;)
# write_csv(go_BP_df, &quot;results/GSEA_analysis_BP.csv&quot;)</code></pre>
<pre class="r"><code># --- Visualization ---
# Dotplot showing activated and repressed pathways
dotplot_gsea &lt;- dotplot(
  gsea_BP, 
  showCategory = 10,       
  split = &quot;.sign&quot;,          
  font.size = 6            
) + 
  facet_grid(.~.sign) +    
  ggtitle(&quot;GSEA analysis&quot;) +
  theme_classic()

print(dotplot_gsea)</code></pre>
<p><img src="figure/figure2_GSEA.Rmd/visualization-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># NES plot for top enriched terms
top_results &lt;- go_BP_df[1:11, ]
library(stringr)
top_results$Description &lt;- str_wrap(top_results$Description, width = 40)

nes_plot &lt;- ggplot(top_results, aes(x = NES, y = reorder(Description, NES))) +
  geom_point(aes(color = NES, size = setSize)) +
  scale_color_gradient2(
    low = ulcer_colors[2],  
    mid = &quot;white&quot;,           
    high = ulcer_colors[1],   
    midpoint = 0,           
    name = &quot;NES Score&quot;
  ) +
  labs(
    x = &quot;Normalized Enrichment Score&quot;,
    y = NULL,
    title = &quot;Gene Set Enrichment Analysis (GSEA)&quot;,
    subtitle = &quot;Top enriched GO terms&quot;,
    size = &quot;Gene Set Size&quot;
  ) +
  theme_classic() +
  theme(
    axis.text.y = element_text(size = 10),
    axis.text.x = element_text(size = 11),
    axis.title.x = element_text(size = 14),
    plot.title = element_text(size = 16, hjust = 0.5)
  )

print(nes_plot)</code></pre>
<p><img src="figure/figure2_GSEA.Rmd/visualization-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># GSEA enrichment plot - most up and down-regulated gene sets
top_indices &lt;- c(which.max(gsea_BP@result$NES), which.min(gsea_BP@result$NES))
enrichment_plots &lt;- gseaplot2(
  gsea_BP, 
  geneSetID = top_indices,
  color = ulcer_colors)

print(enrichment_plots)</code></pre>
<p><img src="figure/figure2_GSEA.Rmd/visualization-3.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="pathway-analysis" class="section level2">
<h2>Pathway Analysis</h2>
<pre class="r"><code># Data Preparation

# Extract t-statistic 
gene_stats &lt;- ranked_idmatch[, 5] 
names(gene_stats) &lt;- ranked_idmatch$external_gene_name

# Get the PROGENy weight matrix
progeny_matrix &lt;- getModel(organism = &quot;Human&quot;, top = 100)

# Find common genes between data and the model
common_genes &lt;- intersect(names(gene_stats), rownames(progeny_matrix))

# Subset genes
progeny_matrix_subset &lt;- progeny_matrix[common_genes, , drop = FALSE]
gene_stats_subset &lt;- gene_stats[common_genes]

# Calculate pathway scores using mt multiplication
pathway_scores &lt;- t(progeny_matrix_subset) %*% gene_stats_subset

# Convert to tidy df
progeny_df &lt;- data.frame(
  pathway = rownames(pathway_scores),
  score = as.numeric(pathway_scores),
  stringsAsFactors = FALSE
)

# Normalize (Z-score) 
progeny_df$normalized_score &lt;- scale(progeny_df$score)[,1]

# Sort by absolute score to keep most important pathways at top
progeny_df &lt;- progeny_df %&gt;%
  arrange(desc(abs(normalized_score)))


# --- Visualization ---

progeny_plot &lt;- ggplot(progeny_df, aes(x = reorder(pathway, normalized_score), 
                                       y = normalized_score, 
                                       fill = normalized_score &gt; 0)) +
  geom_bar(stat = &quot;identity&quot;, color = &quot;black&quot;, width = 0.7) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, color = &quot;gray50&quot;) +
  scale_fill_manual(values = c(&quot;#00AFBB&quot;, &quot;#bb0c00&quot;), 
                   labels = c(&quot;Inhibited&quot;, &quot;Activated&quot;),
                   name = &quot;Pathway Status&quot;) +
  coord_flip() +
  labs(x = &quot;Signaling Pathway&quot;, 
       y = &quot;Normalized Activity Score&quot;, 
       title = &quot;PROGENy Pathway Analysis - Ulcerated vs Non-ulcerated tumors&quot;) +
  theme_classic() +
  theme(
    axis.title = element_text(size = 14, face = &quot;bold&quot;),
    axis.text = element_text(size = 12),
    axis.text.y = element_text(face = &quot;bold&quot;, color = &quot;black&quot;),
    plot.title = element_text(size = 16, face = &quot;bold&quot;, hjust = 0.5),
    plot.subtitle = element_text(size = 14, hjust = 0.5, margin = margin(b = 20)),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),
    legend.text = element_text(size = 11),
    legend.position = &quot;bottom&quot;,
    plot.margin = margin(t = 20, r = 20, b = 20, l = 20)
  )

progeny_plot</code></pre>
<p><img src="figure/figure2_GSEA.Rmd/progeny-1.png" width="1248" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;figures/PROGENy_pathway_activity.pdf&quot;, 
#       progeny_plot, 
#       width = 12, 
#       height = 10, 
#       dpi = 300)</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=es_MX.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=es_MX.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=es_MX.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=es_MX.UTF-8 LC_IDENTIFICATION=C       

time zone: America/Mexico_City
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] here_1.0.1             decoupleR_2.10.0       progeny_1.26.0        
 [4] org.Hs.eg.db_3.19.1    AnnotationDbi_1.66.0   IRanges_2.38.1        
 [7] S4Vectors_0.42.1       Biobase_2.64.0         BiocGenerics_0.50.0   
[10] biomaRt_2.60.1         DOSE_3.30.5            enrichplot_1.24.4     
[13] clusterProfiler_4.12.6 lubridate_1.9.4        forcats_1.0.0         
[16] stringr_1.5.1          dplyr_1.1.4            purrr_1.0.2           
[19] readr_2.1.5            tidyr_1.3.1            tibble_3.2.1          
[22] ggplot2_3.5.1          tidyverse_2.0.0        workflowr_1.7.1       

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.17.1       jsonlite_1.8.9         
  [4] magrittr_2.0.3          farver_2.1.2            rmarkdown_2.29         
  [7] fs_1.6.5                zlibbioc_1.50.0         vctrs_0.6.5            
 [10] memoise_2.0.1           ggtree_3.12.0           progress_1.2.3         
 [13] htmltools_0.5.8.1       curl_6.0.1              gridGraphics_0.5-1     
 [16] parallelly_1.41.0       sass_0.4.9              bslib_0.8.0            
 [19] plyr_1.8.9              httr2_1.0.7             cachem_1.1.0           
 [22] whisker_0.4.1           igraph_2.1.2            lifecycle_1.0.4        
 [25] pkgconfig_2.0.3         gson_0.1.0              Matrix_1.6-5           
 [28] R6_2.5.1                fastmap_1.2.0           GenomeInfoDbData_1.2.12
 [31] digest_0.6.37           aplot_0.2.4             colorspace_2.1-1       
 [34] patchwork_1.3.0         ps_1.8.1                rprojroot_2.0.4        
 [37] RSQLite_2.3.9           labeling_0.4.3          filelock_1.0.3         
 [40] timechange_0.3.0        httr_1.4.7              polyclip_1.10-7        
 [43] compiler_4.4.0          bit64_4.5.2             withr_3.0.2            
 [46] BiocParallel_1.38.0     viridis_0.6.5           DBI_1.2.3              
 [49] ggforce_0.4.2           R.utils_2.12.3          MASS_7.3-60            
 [52] rappdirs_0.3.3          tools_4.4.0             scatterpie_0.2.4       
 [55] ape_5.8-1               httpuv_1.6.15           R.oo_1.27.0            
 [58] glue_1.8.0              callr_3.7.6             nlme_3.1-165           
 [61] GOSemSim_2.30.2         promises_1.3.2          shadowtext_0.1.4       
 [64] grid_4.4.0              getPass_0.2-4           reshape2_1.4.4         
 [67] fgsea_1.30.0            generics_0.1.3          gtable_0.3.6           
 [70] tzdb_0.4.0              R.methodsS3_1.8.2       data.table_1.16.4      
 [73] hms_1.1.3               xml2_1.3.6              tidygraph_1.3.1        
 [76] XVector_0.44.0          ggrepel_0.9.6           pillar_1.10.0          
 [79] yulab.utils_0.1.8       later_1.4.1             splines_4.4.0          
 [82] tweenr_2.0.3            BiocFileCache_2.12.0    treeio_1.28.0          
 [85] lattice_0.22-5          bit_4.5.0.1             tidyselect_1.2.1       
 [88] GO.db_3.19.1            Biostrings_2.72.1       knitr_1.49             
 [91] git2r_0.33.0            gridExtra_2.3           xfun_0.49              
 [94] graphlayouts_1.2.1      stringi_1.8.4           UCSC.utils_1.0.0       
 [97] lazyeval_0.2.2          ggfun_0.1.8             yaml_2.3.10            
[100] evaluate_1.0.1          codetools_0.2-19        ggraph_2.2.1           
[103] qvalue_2.36.0           ggplotify_0.1.2         cli_3.6.3              
[106] munsell_0.5.1           processx_3.8.4          jquerylib_0.1.4        
[109] Rcpp_1.0.13-1           GenomeInfoDb_1.40.1     dbplyr_2.5.0           
[112] png_0.1-8               parallel_4.4.0          blob_1.2.4             
[115] prettyunits_1.2.0       viridisLite_0.4.2       tidytree_0.4.6         
[118] scales_1.3.0            crayon_1.5.3            rlang_1.1.4            
[121] cowplot_1.1.3           fastmatch_1.1-4         KEGGREST_1.44.1        </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=es_MX.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=es_MX.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=es_MX.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=es_MX.UTF-8 LC_IDENTIFICATION=C       

time zone: America/Mexico_City
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] here_1.0.1             decoupleR_2.10.0       progeny_1.26.0        
 [4] org.Hs.eg.db_3.19.1    AnnotationDbi_1.66.0   IRanges_2.38.1        
 [7] S4Vectors_0.42.1       Biobase_2.64.0         BiocGenerics_0.50.0   
[10] biomaRt_2.60.1         DOSE_3.30.5            enrichplot_1.24.4     
[13] clusterProfiler_4.12.6 lubridate_1.9.4        forcats_1.0.0         
[16] stringr_1.5.1          dplyr_1.1.4            purrr_1.0.2           
[19] readr_2.1.5            tidyr_1.3.1            tibble_3.2.1          
[22] ggplot2_3.5.1          tidyverse_2.0.0        workflowr_1.7.1       

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.17.1       jsonlite_1.8.9         
  [4] magrittr_2.0.3          farver_2.1.2            rmarkdown_2.29         
  [7] fs_1.6.5                zlibbioc_1.50.0         vctrs_0.6.5            
 [10] memoise_2.0.1           ggtree_3.12.0           progress_1.2.3         
 [13] htmltools_0.5.8.1       curl_6.0.1              gridGraphics_0.5-1     
 [16] parallelly_1.41.0       sass_0.4.9              bslib_0.8.0            
 [19] plyr_1.8.9              httr2_1.0.7             cachem_1.1.0           
 [22] whisker_0.4.1           igraph_2.1.2            lifecycle_1.0.4        
 [25] pkgconfig_2.0.3         gson_0.1.0              Matrix_1.6-5           
 [28] R6_2.5.1                fastmap_1.2.0           GenomeInfoDbData_1.2.12
 [31] digest_0.6.37           aplot_0.2.4             colorspace_2.1-1       
 [34] patchwork_1.3.0         ps_1.8.1                rprojroot_2.0.4        
 [37] RSQLite_2.3.9           labeling_0.4.3          filelock_1.0.3         
 [40] timechange_0.3.0        httr_1.4.7              polyclip_1.10-7        
 [43] compiler_4.4.0          bit64_4.5.2             withr_3.0.2            
 [46] BiocParallel_1.38.0     viridis_0.6.5           DBI_1.2.3              
 [49] ggforce_0.4.2           R.utils_2.12.3          MASS_7.3-60            
 [52] rappdirs_0.3.3          tools_4.4.0             scatterpie_0.2.4       
 [55] ape_5.8-1               httpuv_1.6.15           R.oo_1.27.0            
 [58] glue_1.8.0              callr_3.7.6             nlme_3.1-165           
 [61] GOSemSim_2.30.2         promises_1.3.2          shadowtext_0.1.4       
 [64] grid_4.4.0              getPass_0.2-4           reshape2_1.4.4         
 [67] fgsea_1.30.0            generics_0.1.3          gtable_0.3.6           
 [70] tzdb_0.4.0              R.methodsS3_1.8.2       data.table_1.16.4      
 [73] hms_1.1.3               xml2_1.3.6              tidygraph_1.3.1        
 [76] XVector_0.44.0          ggrepel_0.9.6           pillar_1.10.0          
 [79] yulab.utils_0.1.8       later_1.4.1             splines_4.4.0          
 [82] tweenr_2.0.3            BiocFileCache_2.12.0    treeio_1.28.0          
 [85] lattice_0.22-5          bit_4.5.0.1             tidyselect_1.2.1       
 [88] GO.db_3.19.1            Biostrings_2.72.1       knitr_1.49             
 [91] git2r_0.33.0            gridExtra_2.3           xfun_0.49              
 [94] graphlayouts_1.2.1      stringi_1.8.4           UCSC.utils_1.0.0       
 [97] lazyeval_0.2.2          ggfun_0.1.8             yaml_2.3.10            
[100] evaluate_1.0.1          codetools_0.2-19        ggraph_2.2.1           
[103] qvalue_2.36.0           ggplotify_0.1.2         cli_3.6.3              
[106] munsell_0.5.1           processx_3.8.4          jquerylib_0.1.4        
[109] Rcpp_1.0.13-1           GenomeInfoDb_1.40.1     dbplyr_2.5.0           
[112] png_0.1-8               parallel_4.4.0          blob_1.2.4             
[115] prettyunits_1.2.0       viridisLite_0.4.2       tidytree_0.4.6         
[118] scales_1.3.0            crayon_1.5.3            rlang_1.1.4            
[121] cowplot_1.1.3           fastmatch_1.1-4         KEGGREST_1.44.1        </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
